src/ed_object_models/spawn_sdf_in_gazebo.py

```diff
@@ -40,8 +40,13 @@ def from_yaml(yaml_path):
         # Define object pose
         object_pose = Pose()
         object_pose.position = Point(item['x'], item['y'], item['z'])
-        if {'roll', 'pitch', 'yaw'} <= set(item):    # Check if euler angle fields exist
-            object_pose.orientation = Quaternion(*eul2quat(item['roll'], item['pitch'], item['yaw']))
+        if not 'roll' in item:     # Check if roll is defined.
```

Comment:
Not very nice to edit the yaml object. Initialize new parameters, fill these if in yaml object.

Reply:
It loads the yaml in as a dict, so adding keys should only affect the scope of this function. And adding intermediate parameters to me feels unnecessary.

Reply:
I know you are not editing the file, but it is not very clean in my opinion. What do you think @alberth.

Reply:
I think the basic problem is that modifying the yaml file-data blurs the distinction between what's in the file and how it's interpreted.

Right now, there is indeed no problem as you're loading the file locally, but if loading the file is ever moved outside this function (yaml file is needed elsewhere too, or merged into some other data), you'll be modifying file-data non-locally. The person doing that split might miss you changing the file-data, and then you have a nice hard to find bug where some code changes a loaded file.

You don't need additional variables to fix this, just use `dict.get` with a default value `0`, that is, `item.get('roll', 0)`. Same for the other two parameters.

Reply:
This indeed looks the best way to go.\